upcloud/load_balancer.go

```diff
@@ -4,72 +4,238 @@ import (
+	LoadBalancerOperationalStateDeleteDNS     LoadBalancerOperationalState = "delete-dns"
+	LoadBalancerOperationalStateDeleteNetwork LoadBalancerOperationalState = "delete-network"
+	LoadBalancerOperationalStateDeleteServer  LoadBalancerOperationalState = "delete-server"
+	LoadBalancerOperationalStateDeleteService LoadBalancerOperationalState = "delete-service"
+
+	LoadBalancerMatcherTypeSrcIP        LoadBalancerMatcherType = "src_ip "
+	LoadBalancerMatcherTypeSrcPort      LoadBalancerMatcherType = "src_port"
+	LoadBalancerMatcherTypeBodySize     LoadBalancerMatcherType = "body_size"
+	LoadBalancerMatcherTypePath         LoadBalancerMatcherType = "path"
+	LoadBalancerMatcherTypeURL          LoadBalancerMatcherType = "url"
+	LoadBalancerMatcherTypeURLQuery     LoadBalancerMatcherType = "url_query"
+	LoadBalancerMatcherTypeHost         LoadBalancerMatcherType = "host"
+	LoadBalancerMatcherTypeHTTPMethod   LoadBalancerMatcherType = "http_method"
+	LoadBalancerMatcherTypeCookie       LoadBalancerMatcherType = "cookie"
+	LoadBalancerMatcherTypeHeader       LoadBalancerMatcherType = "header"
+	LoadBalancerMatcherTypeURLParam     LoadBalancerMatcherType = "url_param"
+	LoadBalancerMatcherTypeNumMembersUP LoadBalancerMatcherType = "num_members_up"
+
+	LoadBalancerActionTypeUseBackend   LoadBalancerActionType = "use_backend"
+	LoadBalancerActionTypeTCPReject    LoadBalancerActionType = "tcp_reject"
+	LoadBalancerActionTypeHTTPReturn   LoadBalancerActionType = "http_return"
+	LoadBalancerActionTypeHTTPRedirect LoadBalancerActionType = "http_redirect"
+
+	LoadBalancerStringMatcherMethodExact     LoadBalancerStringMatcherMethod = "exact"
+	LoadBalancerStringMatcherMethodSubstring LoadBalancerStringMatcherMethod = "substring"
+	LoadBalancerStringMatcherMethodRegexp    LoadBalancerStringMatcherMethod = "regexp"
+	LoadBalancerStringMatcherMethodStarts    LoadBalancerStringMatcherMethod = "starts"
+	LoadBalancerStringMatcherMethodEnds      LoadBalancerStringMatcherMethod = "ends"
+	LoadBalancerStringMatcherMethodDomain    LoadBalancerStringMatcherMethod = "domain"
+	LoadBalancerStringMatcherMethodIP        LoadBalancerStringMatcherMethod = "ip"
+	LoadBalancerStringMatcherMethodExists    LoadBalancerStringMatcherMethod = "exists"
+
+	LoadBalancerHTTPMatcherMethodGet     LoadBalancerHTTPMatcherMethod = "GET"
+	LoadBalancerHTTPMatcherMethodHead    LoadBalancerHTTPMatcherMethod = "HEAD"
+	LoadBalancerHTTPMatcherMethodPost    LoadBalancerHTTPMatcherMethod = "POST"
+	LoadBalancerHTTPMatcherMethodPut     LoadBalancerHTTPMatcherMethod = "PUT"
+	LoadBalancerHTTPMatcherMethodPatch   LoadBalancerHTTPMatcherMethod = "PATCH"
+	LoadBalancerHTTPMatcherMethodDelete  LoadBalancerHTTPMatcherMethod = "DELETE"
+	LoadBalancerHTTPMatcherMethodConnect LoadBalancerHTTPMatcherMethod = "CONNECT"
+	LoadBalancerHTTPMatcherMethodOptions LoadBalancerHTTPMatcherMethod = "OPTIONS"
+	LoadBalancerHTTPMatcherMethodTrace   LoadBalancerHTTPMatcherMethod = "TRACE"
```

Comment:
To be frank, I'm not exactly sure if those need to be constants.
HTTP methods are well known and stable, just using plain strings is imo enough and readable. That being said, it doesn't hurt to have them

Reply:
I thought that this way user can directly see what methods `matcher` supports.